pkg/controller/servingcert/controller/secret_updating_controller.go

```diff
@@ -69,70 +72,134 @@ func (sc *serviceServingCertUpdateController) Key(namespace, name string) (metav
 
 func (sc *serviceServingCertUpdateController) Sync(obj metav1.Object) error {
 	sharedSecret := obj.(*v1.Secret)
+	service := sc.getServiceForSecret(sharedSecret)
+	if service == nil {
+		return nil
+	}
 
-	regenerate, service := sc.requiresRegeneration(sharedSecret)
-	if !regenerate {
+	if !isSecretValidForService(service, sharedSecret) {
 		return nil
 	}
 
 	// make a copy to avoid mutating cache state
 	secretCopy := sharedSecret.DeepCopy()
 
-	if err := toRequiredSecret(sc.dnsSuffix, sc.ca, service, secretCopy); err != nil {
-		return err
+	update := false
+	if sc.requiresRegeneration(service, sharedSecret, sc.minTimeLeftForCert) {
+		// Regenerate the secret
+		if err := toRequiredSecret(sc.dnsSuffix, sc.ca, service, secretCopy); err != nil {
+			return err
+		}
+		update = true
```

Comment:
@stlaz you meant I should update and return here? I do see that.. I'm sorry, this PR is getting difficult to manage.. must merge this today : (

Reply:
nm I see now, it was above.. I commented there.